android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -1147,6 +1148,16 @@ && getActivityTab() != null && getActivityTab().getUrl().getSpec() != null
             ChromeSharedPreferences.getInstance()
                     .writeBoolean(BravePrivacySettings.PREF_APP_LINKS_RESET, false);
         }
+
+        if (PackageUtils.isFirstInstall(this)
```

Comment:
let's create a `boolean for is first install` and call `PackageUtils.isFirstInstall(this)` only once and assign it to a temp variable inside `finishNativeInitialization`. We call it 5 times already from what I see in that function and that PR adds one more call to it. It's an expensive call to get a package info.